drivers/kubevirt/kubevirt.go

```diff
@@ -507,3 +511,181 @@ func AddPVCsToVirtualMachine(vm kubevirtv1.VirtualMachine, pvcs []*corev1.Persis
+}
+
+func getVMPod(appCtx *scheduler.Context, vol *volume.Volume) (*corev1.Pod, error) {
+	pods, err := core.Instance().GetPodsUsingPV(vol.ID)
+	if err != nil {
+		return nil, fmt.Errorf("failed to get pods for volume %s of context %s: %w", vol.ID, appCtx.App.Key, err)
+	}
+
+	var found corev1.Pod
+	for _, pod := range pods {
+		if pod.Labels["kubevirt.io"] == "virt-launcher" && pod.Status.Phase == corev1.PodRunning {
+			if found.Name != "" {
+				// there should be only one VM pod in the running state (otherwise live migration is in progress)
+				return nil, fmt.Errorf("more than 1 KubeVirt pods (%s, %s) are in running state for volume %s",
+					found.Name, pod.Name, vol.ID)
+			}
+			found = pod
+		}
+	}
+	if found.Name == "" {
+		return nil, fmt.Errorf("failed to find a running pod for volume %s", vol.ID)
+	}
+	return &found, nil
+}
+
+func VerifyBindMount(vmi *scheduler.Context, driver scheduler.Driver, volumeDriver volume.Driver) (bool, error) {
+	vols, err := driver.GetVolumes(vmi)
+	if err != nil {
+		return false, err
+	}
+	if len(vols) == 0 {
+		return false, fmt.Errorf("no volumes found for VM [%s] in namespace [%s]", vmi.App.Key, vmi.App.NameSpace)
+	}
+
+	// Get the node where pod is scheduled
+	vms, err := GetAllVMsFromScheduledContexts([]*scheduler.Context{vmi})
+	if err != nil {
+		return false, err
+	}
+
+	vm := vms[0]
+	log.InfoD("pod name: %s, namespace: %s", vmi.App.Key, vmi.App.NameSpace)
+
+	vmNodeName, err := GetNodeOfVM(vm)
+	if err != nil {
+		return false, err
+	}
+	log.Infof("Pod [%s] is deployed on node [%s]", vmi.App.Key, vmNodeName)
+
+	// Keep a track of replicaset and consider this as the source of truth
+	volInspect, err := volumeDriver.InspectVolume(vols[0].ID)
+	if err != nil {
+		return false, fmt.Errorf("failed to inspect volume [%s]: %w", vols[0].ID, err)
+	}
+	globalReplicSet := volInspect.ReplicaSets
+	log.InfoD("Length of replicaset: %d", len(globalReplicSet))
+
+	// Check if the volume is mounted on the node
+	for _, vol := range vols {
+		// Check if the volume is mounted on the node
+		t := func() (interface{}, bool, error) {
+			volInspect, err := volumeDriver.InspectVolume(vol.ID)
+			if err != nil {
+				return "", false, fmt.Errorf("failed to inspect volume [%s]: %w", vol.ID, err)
+			}
+			nodeIpAttachedOn := volInspect.AttachedOn
+			nodeNameAttachedOn, err := node.GetNodeByIP(nodeIpAttachedOn)
+			if err != nil {
+				return "", false, fmt.Errorf("failed to get node name by IP [%s]: %w", nodeIpAttachedOn, err)
+			}
+			log.Infof("Volume [%s] is attached on node [%s]", vol.ID, nodeNameAttachedOn.Name)
+			if err != nil {
+				return "", false, fmt.Errorf("failed to check if volume [%s] is mounted on node [%s]: %w", vol.ID, vmNodeName, err)
+			}
+			if !(nodeNameAttachedOn.Name == vmNodeName) {
+				return "", true, fmt.Errorf("volume [%s] is not mounted on node [%s]", vol.ID, vmNodeName)
+			}
+			return "", false, nil
+		}
+		_, err = task.DoRetryWithTimeout(t, 1*time.Minute, 10*time.Second)
```

Comment:
So in case there is a live migration the volumes and the vm's will not be present in the same node then this would fail. So according to @dbhatnagar-px we can have a different function to check for rebalance of volumes and see if the volume gets moved to the node where vm is present. Maybe we can handle this there Removing retry here.